### PR TITLE
Migrate content from Wiki to docset

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes AB#1957454 - Migrate content from Wiki to docset

## PR Context

Check the boxes below to indicate the content affected by this PR.

**Repository or docset configuration**
- [ ] Repo documentation and configuration (.git/.github/.vscode etc.)
- [ ] Docs build files (.openpublishing.* and build scripts)
- [ ] Docset configuration (docfx.json, mapping, bread, module folder)

**Conceptual documentation**
- [x] Files in docs-conceptual

**Cmdlet reference & about_ topics**
When changing **cmdlet reference** or **about_ topics**, the changes should be copied to all
relevant versions. Check the boxes below to indicate the versions affected by this change.

- [x] Preview content
- [x] Version 7.2 content
- [x] Version 7.0 content
- [x] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _main_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**. If the PR is work in progress,
  please add the prefix `WIP:` or `[WIP]` to the beginning of the title and remove the prefix when
  the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Case-Sensitivity.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+## On all systems
+
+- PowerShell variables are case-insensitive
+
+  Variable names have no interaction between them and the underlying operating system. PowerShell
+  treats them case-insensitively.
+
+- Module names are case-insensitive (with exceptions)
+
+  The _name_ of the module a pure PowerShell concept and treated case-insensitively. However, there
```
Comment (michaeltlombardi) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Case-Sensitivity.md:34]:
```suggestion
  The _name_ of the module is purely a PowerShell concept and treated case-insensitively. However, there
```

Comment (michaeltlombardi) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Case-Sensitivity.md:34]: I believe this change will be needed across versions.

File: reference/docs-conceptual/learn/deep-dives/avoid-using-invoke-expression.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+description: This article explains why Invoke-Expression should only be used as a last resort.
+ms.custom: wiki-migration
+ms.date: 06/06/2022
+title: Avoid using Invoke-Expression
+---
+# Avoid using Invoke-Expression
+
+The `Invoke-Expression` cmdlet should only be used as a _last resort_. In most scenarios, safer and
+more robust alternatives are available. Forums like Stack Overflow are riddled with examples of
+`Invoke-Expression` misuse. Also note that **PSScriptAnalyzer** has a rule for this. For more
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/deep-dives/avoid-using-invoke-expression.md:11]: Slight reword suggestion to sound more like the rest of our docs.
```suggestion
more robust alternatives are available. Forums like Stack Overflow are filled with examples of
`Invoke-Expression` misuse. Also note that **PSScriptAnalyzer** has a rule for this. For more
```

File: reference/docs-conceptual/learn/deep-dives/avoid-using-invoke-expression.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+information, see
+[AvoidUsingInvokeExpression](/powershell/utility-modules/psscriptanalyzer/rules/avoidusinginvokeexpression).
+
+Carefully consider the security implications. When a string from an untrusted source such as user
+input is passed directly to `Invoke-Expression`, arbitrary commands can be executed. Always consider
+a different, usually more robust and secure solution first.
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/deep-dives/avoid-using-invoke-expression.md:17]: Slight reword to drop "usually" here as that's implied from above, I think.

```suggestion
a different, more robust and secure solution first.
```

File: reference/docs-conceptual/learn/deep-dives/avoid-using-invoke-expression.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  should be discouraged.
+
+- **Testing for parsing errors.** The PowerShell team tests for parse errors in the source code using
+  `Invoke-Expression` because that's the only way to turn a parse-time error into a runtime one.
+
+## Conclusion
+
+The PowerShell team testing PowerShell has a reason for `Invoke-Expression`. And as an interpreted
+language, PowerShell must have a way to dynamically run itself. But there's reason to use
+`Invoke-Expression` in a production environment.
```
Comment (michaeltlombardi) [inline reference/docs-conceptual/learn/deep-dives/avoid-using-invoke-expression.md:50]: These last sentences seem a little off to me, especially the last sentence. I think the first two sentences might more properly belong in the section above about how/why the PowerShell team uses `Invoke-Expression`.

The last sentence could maybe be rewritten to something like:

> While there may be good reasons to use `Invoke-Expression` in a production environment, those uses should be kept to a minimum in count and scope of what they execute. Where possible, you should first explore safer alternatives.

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]

**Review by michaeltlombardi — APPROVED** [review]

File: reference/docs-conceptual/learn/deep-dives/avoid-using-invoke-expression.md (hunk 4 above)
Comment (sdwheeler) [reply reference/docs-conceptual/learn/deep-dives/avoid-using-invoke-expression.md]: Fixed

**Pull request merged**